src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -1115,17 +1118,81 @@ private void detectAnomaly(
             }
         }
 
+        String user;
+        List<String> roles;
+        if (adTask.getUser() == null) {
+            user = "";
+            roles = settings.getAsList("", ImmutableList.of("all_access", "AmazonES_all_access"));
```

Comment:
Why is it necessary to add these admin roles? The only other place I see it used is AnomalyDetectorJobRunner.

Reply:
It's possible that user create domain with security disabled, then enable security. This is for BWC, for old detectors which created when security disabled, the user in AD task will be null.